06_gpu_and_ml/text-to-video/mochi.py

```diff
@@ -28,338 +27,147 @@
 
 # ## Setting up the environment for Mochi
 
-# We start by defining the environment the model runs in.
-# We'll need the [full CUDA toolkit](https://modal.com/docs/guide/cuda),
-# [Flash Attention](https://arxiv.org/abs/2205.14135) for fast attention kernels,
-# and the Mochi model code.
-
-import json
-import os
-import tempfile
-import time
-from pathlib import Path
+# At the time of writing, Mochi is supported natively in the [Diffusers](https://github.com/huggingface/diffusers) library.
+# We'll install install diffusers from source to get these latest features.
 
 import modal
 
-MINUTES = 60
-HOURS = 60 * MINUTES
-
+import time
+import string
+from pathlib import Path
 
-cuda_version = "12.3.1"  # should be no greater than host CUDA version
-flavor = "devel"  #  includes full CUDA toolkit
-os_version = "ubuntu22.04"
-tag = f"{cuda_version}-{flavor}-{os_version}"
+app = modal.App()
 
-image = (
-    modal.Image.from_registry(f"nvidia/cuda:{tag}", add_python="3.11")
-    .entrypoint([])
-    .apt_install("git", "ffmpeg")
-    .pip_install("torch==2.4.0", "packaging", "ninja", "wheel", "setuptools")
-    .pip_install("flash-attn==2.6.3", extra_options="--no-build-isolation")
+image = (modal.Image.debian_slim()
+    .apt_install("git")
+    .pip_install("torch", "accelerate", "hf-transfer", "sentencepiece", "imageio[ffmpeg]")
     .pip_install(
-        "git+https://github.com/genmoai/models.git@075b6e36db58f1242921deff83a1066887b9c9e1"
-    )
+        "git+https://github.com/huggingface/diffusers",
```

Comment:
Why do we need to install diffusers from `git`? Did they not cut a release with Mochi yet? If so, we need to pin the revision, as we did with `genmoai/models`, to avoid surprises.

We also want to do our `pip_install`s all at once when installing ML libraries -- `pip` doesn't check package compat across install command invocations and will happily mess up your environment.

Regardless, we want to include that install instead

Reply:
no release cut yet! pinning revision (both model and git hash)